src/main/java/com/github/games647/flexiblelogin/storage/Database.java

```diff
@@ -195,6 +199,29 @@ protected int getRegistrationsCount(InetAddress ip, StatementConsumer ipSerializ
 
         return -1;
     }
+    
+    protected Set<Account> getAccountsByIp(InetAddress ip, StatementConsumer idSetter) {
+        try (Connection conn = dataSource.getConnection();
+             PreparedStatement stmt = conn.prepareStatement("SELECT * FROM " + tableName + " WHERE IP=?")) {
+            idSetter.accept(stmt);
+
+            ImmutableSet.Builder<Account> accountsBuilder = ImmutableSet.builder();
+            try (ResultSet resultSet = stmt.executeQuery()) {
+                do {
+                    Optional<Account> optAccount = parseLoadResult(resultSet);
+                    if(optAccount.isPresent()) {
```

Comment:
Not actually required to be changed, just a hint: This could be simplified to `parseLoadResult(resultSet).ifPresent(accountsBuilder::add);`